test/fakeintake/client/client.go

```diff
@@ -368,12 +369,27 @@ func (c *Client) FilterMetrics(name string, options ...MatchOpt[*aggregator.Metr
 	if err != nil {
 		return nil, err
 	}
+	return filterPayload(metrics, options...)
+}
+
+// FilterCheckRuns fetches fakeintake on `/api/v1/check_run` endpoint and returns
+// metrics matching `name` and any [MatchOpt](#MatchOpt) options
+func (c *Client) FilterCheckRuns(name string, options ...MatchOpt[*aggregator.CheckRun]) ([]*aggregator.CheckRun, error) {
+	checkRuns, err := c.GetCheckRun(name)
+	if err != nil {
+		return nil, err
+	}
+	return filterPayload(checkRuns, options...)
+}
+
+// filterPayload returns payloads matching any [MatchOpt](#MatchOpt) options
+func filterPayload[T aggregator.PayloadItem](payloads []T, options ...MatchOpt[T]) ([]T, error) {
```

Comment:
It would look more organized to have all the public specialized `FilterMetrics`, `FilterCheckRuns`, `FilterLogs`, etc. functions grouped at the beginning of the file and the private generic `filterPayload` one at the end rather than having it in the middle.

Reply:
Agreed ! 👍